analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/SymbolicExecutionRunner.cs

```diff
@@ -82,25 +82,27 @@ protected override void Initialize(SonarAnalysisContext context)
                     var declaration = (AnonymousFunctionExpressionSyntax)c.Node;
                     if (c.SemanticModel.GetSymbolInfo(declaration).Symbol is { } symbol)
                     {
-                        Analyze(c, declaration.Body, symbol);
+                        Analyze(context, c, declaration.Body, symbol);
                     }
                 },
                 SyntaxKind.AnonymousMethodExpression,
                 SyntaxKind.SimpleLambdaExpression,
                 SyntaxKind.ParenthesizedLambdaExpression);
         }
 
-        private void Analyze<TNode>(SyntaxNodeAnalysisContext context, Func<TNode, CSharpSyntaxNode> getBody) where TNode : SyntaxNode
+        private void Analyze<TNode>(SonarAnalysisContext analysisContext, SyntaxNodeAnalysisContext context, Func<TNode, CSharpSyntaxNode> getBody) where TNode : SyntaxNode
         {
             if (getBody((TNode)context.Node) is { } body && context.SemanticModel.GetDeclaredSymbol(context.Node) is { } symbol)
             {
-                Analyze(context, body, symbol);
+                Analyze(analysisContext, context, body, symbol);
             }
         }
 
-        private void Analyze(SyntaxNodeAnalysisContext context, CSharpSyntaxNode body, ISymbol symbol)
+        private void Analyze(SonarAnalysisContext analysisContext, SyntaxNodeAnalysisContext context, CSharpSyntaxNode body, ISymbol symbol)
         {
-            var enabledAnalyzers = symbolicExecutionAnalyzerFactory.GetEnabledAnalyzers(context);
+            var isTestProject = analysisContext.IsTestProject(context.Compilation, context.Options);
+            var isScannerRun = analysisContext.IsScannerRun(context.Options);
```

Comment:
I need these two here for the future PR. That's why I don't pass `context` itself.